plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -161,6 +164,68 @@ class="screen-reader-text"
 		return $block_content;
 	}
 
+	/**
+	 * Track feature usage of the Product Collection block within the site editor.
+	 *
+	 * @param int      $post_id  The post ID.
+	 * @param \WP_Post $post     The post object.
+	 *
+	 * @return void
+	 */
+	public function track_collection_instances( $post_id, $post ) {
+
+		if ( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST || ! wc_current_theme_is_fse_theme() ) {
+			return;
+		}
+
+		if ( ! $post instanceof \WP_Post ) {
+			return;
+		}
+
+		// Important: don't track revisions.
+		$post_type = $post->post_type;
+		if ( ! in_array( $post_type, array( 'post', 'page', 'wp_template', 'wp_template_part' ), true ) ) {
+			return;
+		}
```

Comment:
We also need new instances here!

Reply:
Aha! If I understand correctly, we can't use `$update` because we need to track the new instances as well. Means there will be cases when `$update` will be `true` but there will be new instances as well. Right? 

> // Important: don't track revisions.

This comment feel a bit confusing to me. Instead this we can say: "Track only the new instances of the Product Collection block and not the revisions". 
It was confusing for me because I thought we are talking about the revisions of the post. But actually, we are talking about the revisions of the Product Collection block. 

But It maybe just me who got confused with this comment. So, feel free to ignore it if you think it's clear enough. :slightly_smiling_face:

Reply:
The primary need here is to measure adoption, so the plan is to track:
- PC blocks on new (published) entities and;
- PC blocks on every update (published ones again)

So, whether the `$update` is true or false, it doesn't make a difference to this logic. Does that make sense?

Reply:
Thanks for pushing this, @imanish003 I'll improve the comment for more clarity!